workorders.lic

```diff
@@ -207,32 +197,27 @@ class WorkOrders
     end
   end
 
-  def sew_items(info, item, quantity)
-    recipe = @recipes.find { |r| r['name'] == item }
+  def order_yarn(stock_room, stock_needed, stock_number)
+    return unless stock_needed > 0
 
-    if 'What were' == bput('get yarn', 'What were', 'You get')
-      walk_to info['stock-room']
+    stock_needed.times do |_|
+      order_item(stock_room, stock_number)
+      combine_yarn
+    end
+  end
 
-      stock_needed = (quantity * recipe['volume'] / 100.0).ceil
+  def sew_items(info, item, quantity)
+    recipe = @recipes.find { |r| r['name'] == item }
 
-      stock_needed.times do |_|
-        order_stock(info['stock-number'])
-        combine_yarn
-      end
-    else
-      length = bput('count my yarn', 'You count out \d+ yards').scan(/\d+/).first.to_i
-      stock_needed = ((quantity * recipe['volume'] - length) / 100.0).ceil
-      if stock_needed > 0
-        walk_to info['stock-room']
-        stock_needed.times do |_|
-          order_stock(info['stock-number'])
-          combine_yarn
-        end
-      end
-    end
+    existing = if 'What were' == bput('get yarn', 'What were', 'You get')
+                 0
+               else
+                 bput('count my yarn', 'You count out \d+ yards').scan(/\d+/).first.to_i
+             end
```

Comment:
Really not a fan of this formatting, end doesn't seem to line up and it's hard to read.